worlds/civ_6/Locations.py

```diff
@@ -0,0 +1,193 @@
+            self.location_type = CivVICheckType.GOODY
+        elif name.split("_")[0] == "BOOST":
+            self.location_type = CivVICheckType.BOOST
+
+        if self.name in PRIORITY_LOCATIONS:
+            self.progress_type = LocationProgressType.PRIORITY
+        elif self.name in EXCLUDED_LOCATIONS:
+            self.progress_type = LocationProgressType.EXCLUDED
+        else:
+            self.progress_type = LocationProgressType.DEFAULT
+
+        if self.location_type == CivVICheckType.BOOST:
+            boost_data_list = get_boosts_data()
+            boost_data = next((boost for boost in boost_data_list if boost.Type == name), None)
+            if boost_data and boost_data.Classification == "EXCLUDED":
+                self.progress_type = LocationProgressType.EXCLUDED
+
+
+def generate_flat_location_table() -> Dict[str, CivVILocationData]:
+    """
+    Generates a flat location table in the following format:
+    {
+      "TECH_AP_ANCIENT_00": CivVILocationData,
+      "TECH_AP_ANCIENT_01": CivVILocationData,
+      "CIVIC_AP_ANCIENT_00": CivVILocationData,
+      ...
+    }
+    """
+    era_locations = generate_era_location_table()
+    flat_locations = {}
+    for era_type, locations in era_locations.items():
+        for location_id, location_data in locations.items():
+            flat_locations[location_id] = location_data
+    return flat_locations
+
+
+def generate_era_location_table() -> Dict[EraType, Dict[str, CivVILocationData]]:
+    """
+    Uses the data from existing_tech.json to generate a location table in the following format:
+    {
+      "ERA_ANCIENT": {
+        "TECH_AP_ANCIENT_00": CivVILocationData,
+        "TECH_AP_ANCIENT_01": CivVILocationData,
+        "CIVIC_AP_ANCIENT_00": CivVILocationData,
+      },
+      ...
+    }
+    """
+    new_tech_prereqs = get_new_tech_prereqs_data()
+
+    new_techs = get_new_techs_data()
+    era_locations = {}
+    id_base = 0
+# Techs
+    for data in new_techs:
+        era_type = data['EraType']
+        if era_type not in era_locations:
+            era_locations[era_type] = {}
+
+        prereq_data = [
+            item for item in new_tech_prereqs if item['Technology'] == data['Type']]
+
+        era_locations[era_type][data["Type"]] = CivVILocationData(
+            data["Type"], data['Cost'], data['UITreeRow'], id_base, era_type, CivVICheckType.TECH, prereq_data)
+        id_base += 1
+# Civics
+    new_civic_prereqs = get_new_civic_prereqs_data()
+    new_civics = get_new_civics_data()
+
+    for data in new_civics:
+        era_type = data['EraType']
+        if era_type not in era_locations:
+            era_locations[era_type] = {}
+        prereq_data = [
+            item for item in new_civic_prereqs if item['Civic'] == data['Type']]
+        era_locations[era_type][data["Type"]] = CivVILocationData(
+            data["Type"], data['Cost'], data['UITreeRow'], id_base, era_type, CivVICheckType.CIVIC, prereq_data)
+        id_base += 1
```

Comment:
```suggestion
    for data in new_civics:
        era_type = data['EraType']
        if era_type not in era_locations:
            era_locations[era_type] = {}
        prereq_data = [
            item for item in new_civic_prereqs if item["Civic"] == data["Type"]]
        era_locations[era_type][data["Type"]] = CivVILocationData(
            data["Type"], data["Cost"], data["UITreeRow"], id_base, era_type, CivVICheckType.CIVIC, prereq_data)
        id_base += 1
```
Would recommend more consistency with whether you're using ' or " -- sometimes you use both in the same line, even

Reply:
Dangit, yeah I usually trust my formatter to take care of these. It is all over my metroid prime apworld as well lol. Going to update the formatter, thank you for noticing 🙏